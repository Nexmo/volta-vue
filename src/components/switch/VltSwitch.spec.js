import { shallowMount } from '@vue/test-utils';
import { VltSwitch } from '..';

describe('vlt-switch', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltSwitch);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is red', async () => {
    wrapper.setProps({ red: true });
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });

  test('can be checked', async () => {
    await wrapper.find('input').trigger('change');
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted('input')).toEqual([[true]]);
  });
});
