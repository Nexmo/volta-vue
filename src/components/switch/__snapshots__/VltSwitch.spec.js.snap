// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-switch is red 1`] = `
<div
  class="Vlt-switch Vlt-switch--red"
>
  <label>
    <input
      type="checkbox"
    />
    <!-- Without @click.stop, click on switch will trigger 2 click events, one for the input
      and one for span.Vlt-switch__slider -->
    <span
      class="Vlt-switch__slider"
    />
  </label>
</div>
`;

exports[`vlt-switch renders correctly 1`] = `
<div
  class="Vlt-switch"
>
  <label>
    <input
      type="checkbox"
    />
    <!-- Without @click.stop, click on switch will trigger 2 click events, one for the input
      and one for span.Vlt-switch__slider -->
    <span
      class="Vlt-switch__slider"
    />
  </label>
</div>
`;
